File: c/linearSearch.c
/*
*   Linear Search Algorithm in Language C
*
* Author: Elton Fonseca
*/

#include <stdio.h>
#include <string.h>

int search(int *array, int size, int value);

int main(void)
{
    int array[10] = {1,2,3,4,5,6,7,8,9,10};
    
    if(search(array, 10, 9))
        puts("Find!");
    else
        puts("Not Found!") ;
}

int search(int *array, int size, int value)
{
    int i;

    for(i = 0; i < size; i++)
        if(array[i] == value)
            return 1;
    return 0;
}

File: javascript/linearSearch.js
/*
 * Linear Search Algorithm in JavaScript
 * 
 * Author: Elton Fonseca 
 */

function main() 
{
    var array = [0,1,2,3,4,5,6,7,8,9,10];

    if(search(array, 11))
        console.log("Find!");
    else
        console.log("Not Found!");
}

function search(array, value)
{
    for(var i = 0; i < array.length; i++)
    {
        if(value == array[i])
            return true;
    }
    return false;
}
main();